File: PlusOne.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry = 0;
        int size = digits.size()-1;
        int pos = size;
        cout << size+1 << endl;
        
        
        if(digits[size] != 9){//if under nine add 1 and return
            digits[size]+=1;
        }else{
            carry+=1;
            while(carry != 0){
                cout << "pos: " << pos << endl;
                if(pos == -1){
                    digits.push_back(0);
                    for(int x : digits){
                        x = 0;
                    }
                    digits[0] = 1;
                    return digits;
                }
                
                if(digits[pos] == 9){
                    digits[pos] = 0;
                    pos--;
                }else{
                    digits[pos]++;
                    return digits;
                }
                
            }
        
        }
        return digits;

    }
};

File: PlusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        pos = len(digits) -1
        run = True
        
        
        while run:
            if digits[pos] == 9:
                digits[pos] = 0
                pos-=1
            else:
                digits[pos] +=1
                run = False
                
            if pos == -1:
                digits.insert(0,1)
                run = False
                
        return digits